### PR TITLE
Fork Presto ES connector for continued ES 6 support

### PR DESCRIPTION
**Pull request by misterjpapa**
Upgrading the Presto Elasticsearch connector to use Elasticsearch 7.0
libraries will introduce breaking changes for supporting Elasticsearch 6.0
servers due to the lack of backward compatibility in the Elasticsearch 
v7.0 RestHighLevelClient (HLC) and the Types deprecated in APIs in 
ES 7.0 introducing breaking changes to the index creation, put mapping, 
get mapping, put template, get template and get field mappings APIs.

By forking and creating a presto-elasticsearch-6 version of the 
Elasticsearch connector as a new module we can mitigate these 
breaking changes so users can stay using the older connector code if 
they need to stay on the older ES 6.x servers.  As it's own module we can
release this independently and provide instructions for users to retain 
ES6 access. If it goes as a submodule in the main Presto repository it can 
cause issues with CI and the versioning won't make sense since it would 
follow Presto's versioning, when as a standalone plugin it should be 
versioned independently.

The existing presto-elasticsearch module source, tests, and config files 
were copied, renamed to presto-elasticsearch-6 and copied thi new 
repository. Update the name of the plugin to ElasticsearchLegacyPlugin 
and the connector name to elasticsearch_legacy to avoid conflicts.

## Description
Make a copy of the code in the existing presto-elasticsearch
module and rename it to presto-elasticsearch-6. Copy, duplicate, and
rename any references to the presto-elasticsearch module with the
presto-elasticsearch-6 reference. Update the name of the plugin to 
ElasticsearchLegacyPlugin and the connector name to 
elasticsearch_legacy to avoid conflicts. Move this code into new
repository called presto-elasticsearch-6 as it's own independent
module. 

## Motivation and Context
Moving to Elasticsearch 7.0 libraries will introduce breaking changes 
to the Presto Elasticsearch connector with Elasticsearch 6.0 servers 
due to lack of backward compatibility. By forking and creating a 
presto-elasticsearch-6 version of the Elasticsearch connector based
on the current code we can mitigate these breaking changes so 
users can stay using the older connector code if they need to stay on 
the older ES 6.x servers. 
https://github.com/prestodb/presto/issues/23470

## Impact
Presto will provide two Elasticsearch connectors. 
The presto-elasticsearch-6 version of the connector will be for
Elasticsearch v6 servers while the presto-elasticsearch version will be
for Elasticsearch v7 servers.

## Test Plan
Copied the existing 56 tests in the Presto Elasticsearch connector
and made sure they are still running as-is.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Connectors
* Add a presto-elasticsearch-6 module to keep support for a Presto Elasticsearch connector for Elasticsearch 6.0 servers.  Change the plugin name to ElasticsearchLegacyPlugin and connector name to elasticsearch_legacy to avoid conflicting with the existing Elasticsearch plugin. :pr:`23500`

### CODE REVIEW

**Comment by misterjpapa** [comment]
CC @tdcmeehan

**Comment by wanglinsong** [comment]
I think it is better to use plugin name Elasticsearch6, instead of ElasticsearchLegacyPlugin.

**Review by wanglinsong — COMMENTED** [review]
I assume this is the copy from presto repo.
What are the new files in this commit?

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,400 @@
+<?xml version="1.0"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.facebook.presto</groupId>
+        <artifactId>presto-root</artifactId>
+        <version>0.289-SNAPSHOT</version>
+    </parent>
+    <artifactId>presto-elasticsearch-6</artifactId>
+    <description>Presto - Elasticsearch v6 Connector</description>
```
Comment (tdcmeehan) [inline pom.xml:10]: Is it possible to give this a new version, such as `<version>0.0.1</version>`

**Review by tdcmeehan — COMMENTED** [review]
Nice work, thanks for the fork.  Can you add a basic Github actions job to verify the tests don't fail and that the formatting checks succeed?  Something along these lines should work: https://github.com/prestodb/airlift/pull/77/files

**Comment by misterjpapa** [comment]
> I assume this is the copy from presto repo. What are the new files in this commit?

Hi @wanglinsong - no new files. The expectation is that this will be just a copy of the existing ES connector and the code. The new files/changes will go in the presto repo via https://github.com/prestodb/presto/pull/23463.

**Comment by misterjpapa** [comment]
> Nice work, thanks for the fork. Can you add a basic Github actions job to verify the tests don't fail and that the formatting checks succeed? Something along these lines should work: https://github.com/prestodb/airlift/pull/77/files


Will do. I'll see what I can do.

**Comment by misterjpapa** [comment]
> Nice work, thanks for the fork. Can you add a basic Github actions job to verify the tests don't fail and that the formatting checks succeed? Something along these lines should work: https://github.com/prestodb/airlift/pull/77/files

Done.

File: pom.xml (hunk 1 above)
Comment (misterjpapa) [reply pom.xml:10]: I've created a new version called  <version>0.0.1-SNAPSHOT</version>. Please note that version you specified is actually the version of the parent which is the presto-root. That should stay as 0.289-SNAPSHOT.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
Comment (tdcmeehan) [inline .github/workflows/maven.yml]: Can you modify this to `main` so the tests run?

**Review by tdcmeehan — CHANGES_REQUESTED** [review]

File: .github/workflows/maven.yml (hunk 2 above)
Comment (misterjpapa) [reply .github/workflows/maven.yml]: > Can you modify this to `main` so the tests run?

Yikes...sorry that I missed that. It is fixed now.

**Comment by misterjpapa** [comment]
@tdcmeehan  - any thing else you want here?

**Pull request merged**